### ISSUE TITLE
Download Progress Bar

### ISSUE THREAD

**Issue by Actalica** [derrod/legendary#23]
It would make the download progress much easier to understand. You can use rich library for python to make beautiful progress bars.

**Comment by derrod** [comment]
While the current progress indicator is a placeholder and I do plan on replacing it with a nicer one later, I do not quite understand what is hard to understand about a progress percentage?

**Comment by Actalica** [comment]
The precentage is fine but I think it has a few unnecessary details for the average user. You can remove these details and make them optional. Also, it would be easier to understand if you can visualise the progress.

**Comment by artemyto** [comment]
I think it is better to update progress information on the same place, rather than spamming terminal with tons of text. For reference: https://stackoverflow.com/questions/517127/how-do-i-write-output-in-same-place-on-the-console

**Comment by John-Gee** [comment]
Agreed to this, filling up the console with such little information is not helpful.

**Comment by alx365** [comment]
Does the logging library used support this [(link)](https://docs.python.org/3/howto/logging.html)?

**Comment by alx365** [comment]
So, a few days later... 
I've been trying a bit to resize with the terminal, but i've been strugelling A LOT.
It just won't fit with every terminal size. 
i think i'll make it a make it a small progress bar, that almost will never fail:
![image](https://user-images.githubusercontent.com/30902201/96279642-7a9d4300-0fc6-11eb-8ac3-735ed95a4e46.png)

**Comment by zezic** [comment]
Consider using the [tqdm](https://github.com/tqdm/tqdm)

**Comment by nutterthanos** [comment]
> I think it is better to update progress information on the same place, rather than spamming terminal with tons of text. For reference: https://stackoverflow.com/questions/517127/how-do-i-write-output-in-same-place-on-the-console

i would do the same with install/update progress
